### DISCUSSION TITLE
Is there a way to do this in JS other than snippets?

### DISCUSSION THREAD

**Discussion by icarusgk** [zed-industries/zed#20441] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I've seen this feature in Go and Rust while using Zed, is there a way to do it in JavaScript/Typescript?

### Environment

Zed: v0.160.7 (Zed)
OS: macOS 15.1.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help present your vision of the feature

https://github.com/user-attachments/assets/a5612463-115c-4d5a-849c-e86c4b3b1f7c

**Comment by 0x2CA** [comment]
I don't think there is any difference with snippet, if you want to realize it you need to implement this function in lsp, but the extensibility, freedom can't be guaranteed

**Reply by icarusgk** [reply]
I see, does Zed do it at the LSP level too? For Go and Rust in this case.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
